### PR TITLE
Update React and move to optionalDependencies

### PR DESCRIPTION
**Pull request by lemonmade**
This PR updates the React plugin to the most recent version and moves it to `optionalDependencies` to prevent warnings for non-React projects. The only new rule is `jsx-equals-spacing`, which we were already consistently treating as `'never'` unless I'm mistaken.

cc/ @bouk

### CODE REVIEW

**Comment by lemonmade** [comment]
Actually, I don't know if this is what we want — I don't think it's a dependency of this package, it's really just a dependency that the consumer can choose or have or not to have (really, an optional peer dependency). I don't think there's actually a way to declare that, is there?

**Comment by bouk** [comment]
@lemonmade nah optionalDependency is right (it's exactly as you describe, an optional peer dependency). It just makes sure that whatever the upstream consumers sets as the version is in line with what this package needs, so it doesn't break compatibility.

File: package.json [hunk 1]
```diff
@@ -33,12 +33,14 @@
     "babel-eslint": "^4.1.2",
     "eslint": "^1.10.0",
     "eslint-config-shopify": "file:.",
-    "eslint-plugin-react": "^3.15.0",
     "eslint-plugin-shopify": "^2.1.0"
   },
+  "optionalDependencies": {
+    "babel-eslint": "^4.1.2",
```
Comment (bouk) [inline package.json]: Why is this added too?

Also `eslint-plugin-react` should be in devDependencies so it's installed while working on this package

Comment (lemonmade) [inline package.json]: If people use it, I want to make sure they use a version that we know works. Should it not be here?

Comment (lemonmade) [inline package.json]: Do we want the `react` plugin? It's not actually used for any development in this repo, since there's no React being linted here.

**Comment by bouk** [comment]
Lgtm
